lib/src/clojure_lsp/crawler.clj

```diff
@@ -46,34 +38,22 @@
       "Analyzing external classpath"
       progress-token)))
 
-(defn ^:private analyze-external-classpath!
+(defn ^:private classpath-analysis
```

Comment:
Can I get a careful review of this, to make sure this refactoring is correct?

Originally both `analyze-classpath!` and `analyze-external-classpath!` swapped the kondo results into the db, which (I think) was redundant, since one was called by the other. I made it into two functions, one to calculate the kondo results (`classpath-analysis`) and another to swap into the db (`analyze-classpath!`). Does that seem correct?

Reply:
I think they were doing different things originally, `analyze-classpath!`  was saving into db `:analysis` and other related stuff, while `analyze-external-classpath!` was upserting only `:kondo-config`, right?

These 2 functions splitted were a result of previous refactorings, where we had analyze-source-paths calling `analyze-classpath`, I think we could merge those into 1 since it' s only being called to analyze external ones?

Reply:
or if you prefer, have a function to side effect swap and another to return the built map (I think that's what you did right?)

Reply:
It ended up all in one function. It seems easy enough to read, to me, but I'd be happy to separate it if you prefer.